File: frontend/types/analytics.ts
export interface TimeSeriesPoint {
  label: string;
  value: number;
}

export interface NameValuePair {
  name: string;
  value: number;
}

export interface FacilityProcedureVolume {
  facility_id: number;
  facility_name: string;
  city: string;
  GASTROSCOPY: number;
  COLONOSCOPY: number;
  ERCP: number;
  SIGMOIDOSCOPY: number;
  total: number;
}

export interface ReferralBottleneck {
  status: string;
  count: number;
  avg_age_days: number;
}

export interface DiseaseBurden {
  registry_type: string;
  patient_count: number;
}

export interface AnalyticsDashboard {
  total_patients: number;
  total_referrals: number;
  total_procedures: number;
  total_liver_records: number;
  referral_trend: TimeSeriesPoint[];
  facility_procedure_volumes: FacilityProcedureVolume[];
  referral_bottlenecks: ReferralBottleneck[];
  disease_burden: DiseaseBurden[];
  liver_diagnosis_breakdown: NameValuePair[];
  risk_distribution: NameValuePair[];
  procedure_type_breakdown: NameValuePair[];
}

export type MembershipStatus = "ACTIVE" | "PENDING" | "EXPIRED";

export interface Membership {
  id: number;
  user_id: number;
  status: MembershipStatus;
  renewal_date?: string;
  cpd_points_accumulated: number;
  created_at: string;
  updated_at?: string;
  member_name?: string;
  member_email?: string;
  member_role?: string;
}

export interface Guideline {
  id: number;
  title: string;
  category: string;
  published_date: string;
  file_url: string;
  description?: string;
}

export interface Conference {
  id: number;
  title: string;
  location: string;
  event_date: string;
  deadline?: string;
  description?: string;
  registration_url?: string;
  tags: string[];
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List


class TimeSeriesPoint(BaseModel):
    label: str   # e.g. "2024-W23" or "Jun 2024"
    value: int


class NameValuePair(BaseModel):
    name: str
    value: int


class FacilityProcedureVolume(BaseModel):
    facility_id: int
    facility_name: str
    city: str
    GASTROSCOPY: int = 0
    COLONOSCOPY: int = 0
    ERCP: int = 0
    SIGMOIDOSCOPY: int = 0
    total: int = 0


class ReferralBottleneck(BaseModel):
    status: str
    count: int
    avg_age_days: float   # average time a referral has spent in this status


class DiseaseBurden(BaseModel):
    registry_type: str
    patient_count: int


class AnalyticsDashboard(BaseModel):
    # High-level KPIs (reused from legacy summary)
    total_patients: int
    total_referrals: int
    total_procedures: int
    total_liver_records: int
    # Trend: last 12 weeks
    referral_trend: List[TimeSeriesPoint]
    # Facility performance
    facility_procedure_volumes: List[FacilityProcedureVolume]
    # Referral pipeline
    referral_bottlenecks: List[ReferralBottleneck]
    # Disease / registry distribution
    disease_burden: List[DiseaseBurden]
    # Liver diagnosis breakdown
    liver_diagnosis_breakdown: List[NameValuePair]
    # Risk distribution
    risk_distribution: List[NameValuePair]
    # Procedure type breakdown
    procedure_type_breakdown: List[NameValuePair]
